### DISCUSSION TITLE
New Web GUI started

### DISCUSSION THREAD

**Discussion by timothyjoh** [danielmiessler/Fabric#522] [Show and tell]
I have a number of plans for this in the future, but today I put up a pull request #508  which adds a web application that can run locally on your computer and give you access to all the patterns, GPT models and a couple different modes of using it.

![CleanShot 2024-06-03 at 14 58 20@2x](https://github.com/danielmiessler/fabric/assets/1566/96288382-3342-4d3f-810d-7196d0bb042b)

@danielmiessler if you would rather keep this out of the main repo, I would be happy to just have it as a separate tool that is also downloadable.

Future plans: I want to build a UI that allows for custom pattern creation/saving and run it against some document for input, then show the output from the LLM. My desire is to let some people at my company iterate on building patterns that they would need, by having a tool that allows them to iterate over and over while they build the pattern. 

Quick mockup of the next UI to "build new patterns":
![CleanShot 2024-06-03 at 15 04 57@2x](https://github.com/danielmiessler/fabric/assets/1566/3503136b-a25e-473c-8dd6-89f4d0802a9b)

**Comment by lionardben211** [comment]
Very good work. We need that

**Comment by mga-py** [comment]
this would be a beast if combined well with open webui https://docs.openwebui.com/

**Reply by shaneholloman** [reply]
agreed

**Reply by xeniode** [reply]
Yes, glad I found this. I already started doing just that with open webui. @timothyjoh Do you have this as a separate code base?

**Reply by faycalrafa** [reply]
do you guys got any primary release ?

**Comment by ShadowRocket227** [comment]
Amazing, finally we can all use Fabric easily...the only problem i was wondering is: can we still get good markdown output with a UI ? I'd prefer that the Fabric's outputs always be "raw" (a markdown not interpreted)...that would be better for i don't know repository stuff etc.

**Reply by faycalrafa** [reply]
we could have it interpreted and have a copy button to copy the markdown version i think this is better or have an option to toggle between interpreted and not

**Reply by ShadowRocket227** [reply]
I think that would be good, ChatGPT does exactly the same.

**Comment by OriLiMu** [comment]
is this featrue can be used now, cant find related in the help ?

**Comment by xeniode** [comment]
Any updates?

**Comment by diffficult** [comment]
looking forward to any updates on this

**Comment by noamsiegel** [comment]
I made my own fabric front-end app [here](https://github.com/noamsiegel/fabric-app)
